### PR TITLE
'Remove method' confirmation modal

### LINKED ISSUE #1876: [settings] show confirmation modal on payment method removal
## Description
When a payment method is removed from checkout (by clicking the checkbox), we need to display a modal to confirm the merchant's intent.

Suggestion: wrap the confirmation modal and the checkbox is its own component, so make testing easier.

## Acceptance criteria
- When the checkbox is getting un-checked, confirmation modal is displayed
- UI matches designs
- On confirmation, modal is closed and checkbox is un-checked
- On dismiss, modal is closed and checkbox remains checked

### PR DESCRIPTION
**Pull request by Mayisha**
Fixes #1876

### Description
In the Stripe settings page `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe`, checking and unchecking any payment method reflects immediately on click. The changes in this PR introduces a confirmation modal for unchecking a selected method. 

confirmation modal

# Testing instructions
- Enable `_wcstripe_feature_upe_settings`
- Enable UPE
- Go to `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe` and uncheck any checked method. 
- Confirmation modal should open. On clicking `Cancel` should do nothing, on clicking `Remove` should uncheck the payment method.

### CODE REVIEW

**Comment by harriswong** [comment]
👀

File: client/settings/general-settings-section/remove-method-confirmation-modal.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		<ConfirmationModal
+			title={
+				<AlertTitle
+					title={ interpolateComponents( {
+						mixedString: __(
+							'Remove {{methodName /}} from checkout',
+							'woocommerce-gateway-stripe'
+						),
+						components: {
+							methodName: <span>&nbsp;{ label }&nbsp;</span>,
```
Comment (harriswong) [inline client/settings/general-settings-section/remove-method-confirmation-modal.js]: If all we need is a string with a substituted variable, then we may not need an `interpolateComponents`. We can use [sprintf](https://github.com/WordPress/gutenberg/tree/trunk/packages/i18n#sprintf) from `'@wordpress/i18n'`
```
title={ sprintf(
						__(
							'Remove %s from checkout',
							'woocommerce-gateway-stripe'
						),
						label
					) }
```

One benefit is we don't need an extra `<span>` nor `&nbsp;`.

File: client/settings/general-settings-section/remove-method-confirmation-modal.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		>
+			<p>
+				{ interpolateComponents( {
+					mixedString: __(
+						'Are you sure you want to remove {{methodName /}}? Your customers will no longer be able to pay using {{methodName /}}.',
+						'woocommerce-gateway-stripe'
+					),
+					components: {
+						methodName: <span>{ label }</span>,
+					},
```
Comment (harriswong) [inline client/settings/general-settings-section/remove-method-confirmation-modal.js]: Similar to the comment in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1986/files#r717940484, we can use `sprintf` here, something like:
```
{ sprintf(
					__(
						'Are you sure you want to remove %1$s? Your customers will no longer be able to pay using %1$s.',
						'woocommerce-gateway-stripe'
					),
					label
				) }
```

File: client/settings/general-settings-section/payment-methods-list.js [hunk 3]
```diff
@@ -65,45 +70,67 @@ const GeneralSettingsSection = () => {
...
+					<RemoveMethodConfirmationModal
+						method={ modalOpenForMethod }
+						onClose={ () => setIsConfirmationModalOpen( false ) }
+						handleRemove={ () =>
+							handleRemoveMethod( modalOpenForMethod )
+						}
+					/>
+				) }
+			</List>
+		</>
```
Comment (harriswong) [inline client/settings/general-settings-section/payment-methods-list.js]: Looks like `<List></List>` is already wrapping the component. We can probably remove `<></>`.

**Review by harriswong — COMMENTED** [review]
Tested "Remove", "Cancel", "x", all working as expected. I have some minor suggestions, otherwise LGTM 👍

File: client/settings/general-settings-section/remove-method-confirmation-modal.js (hunk 1 above)
Comment (Mayisha) [reply client/settings/general-settings-section/remove-method-confirmation-modal.js]: Thanks for the suggestion :100:

**Comment by Mayisha** [comment]
Addressed the feedbacks :tada:

**Review by harriswong — APPROVED** [review]
Last commit looks good. Tested again by checking/unchecking certain payments, saved, refresh page, still works!
👍  🚢:

**Pull request merged**